csm/core/controllers/version.py

```diff
@@ -0,0 +1,64 @@
+# CORTX-CSM: CORTX Management web and CLI interface.
+# Copyright (c) 2020 Seagate Technology LLC and/or its Affiliates
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as published
+# by the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU Affero General Public License for more details.
+# You should have received a copy of the GNU Affero General Public License
+# along with this program. If not, see <https://www.gnu.org/licenses/>.
+# For any questions about this software or licensing,
+# please email opensource@seagate.com or cortx-questions@seagate.com.
+
+import json
+from marshmallow import Schema, fields, validate
+from marshmallow.exceptions import ValidationError
+from csm.core.controllers.view import CsmView, CsmResponse, CsmAuth
+from cortx.utils.log import Log
+from csm.common.errors import InvalidRequest
+from csm.core.blogic import const
+from csm.core.controllers.rgw.s3.base import S3BaseSchema
+
+class VersionValidationSchema(S3BaseSchema):
+    # Define schema here
+    requires = fields.List(fields.Str, data_key=const.REQUIRES, required=True)
+
+@CsmView._app_routes.view("/api/v2/version/compatibility/{resource}/{resource_id}")
+class VersionCompatibilityView(S3BaseView):
```

Comment:
S3BaseSchema , S3BaseView why we have S3 related code here? Please revamp your classes.

Reply:
updated